### PR TITLE
services/horizon/internal: Create separate db connections for ingestion

### PR DESCRIPTION
**Pull request by tamirms**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Fixes https://github.com/stellar/go/issues/2018

Create separate db connections to the horizon and stellar core databases for ingestion.

### Why

> Currently, all of the Horizon services share the same DB connection pool. In case of increased number of HTTP requests, it's possible that ingestion service will have to wait for an idle connection.

### Known limitations

[N/A]

### CODE REVIEW

File: services/horizon/internal/init.go [hunk 1]
```diff
@@ -19,33 +19,60 @@ import (
...
 
-func mustInitCoreDB(app *App) {
-	session, err := db.Open("postgres", app.config.StellarCoreDatabaseURL)
-	if err != nil {
-		log.Fatalf("cannot open Core DB: %v", err)
+func mustInitHorizonDB(app *App) {
+	maxIdle := app.config.CoreDBMaxIdleConnections
+	maxOpen := app.config.CoreDBMaxOpenConnections
+	if maxIdle >= maxIngestionDBConnections*2 {
+		maxIdle -= maxIngestionDBConnections
```
Comment (abuiles) [inline services/horizon/internal/init.go:39]: could you drop a comment on why do we do this comparison (what does the `2` means for example) and why do we change the value

**Comment by bartekn** [comment]
We can probably merge it into `master` as previous PRs for this release (no breaking changes).

File: services/horizon/internal/init.go [hunk 2]
```diff
@@ -19,33 +19,70 @@ import (
 	"github.com/stellar/go/support/log"
 )
 
-func mustInitHorizonDB(app *App) {
-	session, err := db.Open("postgres", app.config.DatabaseURL)
+const maxIngestionDBConnections = 2
```
Comment (bartekn) [inline services/horizon/internal/init.go:22]: I think it belongs to the `ingest` package. Say, we add a new routine that require an extra DB connection. Would be easier to increment it there.

File: services/horizon/internal/init.go [hunk 3]
```diff
@@ -19,33 +19,70 @@ import (
...
-	if err != nil {
-		log.Fatalf("cannot open Core DB: %v", err)
+func mustInitHorizonDB(app *App) {
+	maxIdle := app.config.HorizonDBMaxIdleConnections
+	maxOpen := app.config.HorizonDBMaxOpenConnections
+	// There are two connection pools, one for serving requests to horizon
+	// and another pool for ingestion. Ideally the total connections in both
+	// pools should be bounded by HorizonDBMaxOpenConnections. But, if the ingestion
+	// pool consumes a significant quota of HorizonDBMaxOpenConnections then we will
+	// allow a total of HorizonDBMaxOpenConnections + maxIngestionDBConnections connections.
```
Comment (bartekn) [inline services/horizon/internal/init.go:42]: I think we should just return an error (maybe at a command validation stage in `root.go`) when ingestion connections is equal/greater than max set by a user. The reason is that user may explicitly set the max number of open connections in Postgres config so these extra 2 connections won't be established (and will probably generate a bunch of errors in the log).

**Pull request merged**